filesystem/fs_interface.go

```diff
@@ -0,0 +1,13 @@
+package filesystem
+
+type FileSystem interface {
```

Comment:
We don't need s3 interface. Just local and HDFS.

This interface is the client in my mind. I'm gonna implement two func:
```
NewHdfsFileSystem()
NewLocalFileSystem()
```